### DISCUSSION TITLE
Ionic Capacitor receive Calls

### DISCUSSION THREAD

**Discussion by danielehrhardt** [ionic-team/capacitor#3163] [Q&A] [answered]
Hi all,

I am currently creating something like the WhatsApp video chat App, my question is the following, if the app is closed on the phone, how to receive the call and show some overlay, iam currently using capacitor with ionic and try to use background Mode but I think it will be killed on someone phones. 

How to make sure to always receive the call? 
Push? (Does not always vibrate, the sound does appear only one time) 
Custom Native Plugin? 

I do not understand how WhatsApp can start itself when a call come in. 

Best
Daniel

**Comment by jcesarmobile** [comment, marked as answer]
Apple created Pushkit for iOS to receive VoIP calls https://developer.apple.com/documentation/pushkit, so you should create a plugin that uses it.
Not sure about Android.
